crates/task-impls/src/vid.rs

```diff
@@ -355,6 +364,53 @@ where
                     ))
                     .await;
             }
+
+            HotShotEvent::TransactionsSequenced(payload, _metadata, view) => {
+                // encode transactions for disperse calculation
+                let encoded_txns = match payload.encode() {
+                    Ok(encoded) => encoded,
+                    Err(e) => {
+                        error!("Failed to encode the block payload: {:?}.", e);
+                        return None;
+                    }
+                };
+
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
+                let srs = test_srs(NUM_STORAGE_NODES);
+                // TODO We are using constant numbers for now, but they will change as the quorum size
+                // changes.
+                // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
+                let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
+                let vid_disperse = vid
+                    .disperse(encoded_txns.into_iter().collect::<Vec<u8>>())
+                    .unwrap();
+
+                // TODO never clone a block
+                // https://github.com/EspressoSystems/HotShot/issues/1858
+                self.event_stream
+                    .publish(HotShotEvent::BlockReady(
+                        Proposal {
+                            data: VidDisperse {
+                                view_number: view,
+                                payload_commitment: payload.commit(),
+                                shares: vid_disperse.shares,
+                                common: vid_disperse.common,
+                            },
+                            // TODO (Keyao) This is also signed in DA task.
```

Comment:
Nit: I think we could remove this comment since the signing isn't the same as what's in the DA task anymore.